lib/fw.c

```diff
@@ -1067,6 +1070,25 @@ static int switchtec_fw_part_info_gen4(struct switchtec_dev *dev,
 	case SWITCHTEC_FW_PART_ID_G4_NVLOG:
 		part_info = &all->nvlog;
 		break;
+	case SWITCHTEC_FW_PART_ID_G4_SEEPROM:
+		if (switchtec_gen(dev) <= SWITCHTEC_GEN4)
+			return 0;
```

Comment:
Wait.... this is only applicable to gen5?? It might be better written as < SWITCHTEC_GEN5 then...

And shouldn't we be setting all the inf-> things to invalid values or something?

Reply:
The caller function already initializes all fields to 0 (valid = false, active=false) so it should suffice.

Reply:
Oh, then maybe we don't need to set the values below to zero.... It's a bit confusing in that way.